### PR TITLE
Fix initialization checks

### PR DESCRIPTION
**Pull request by neokore**

### CODE REVIEW

File: packages/custom-storage/src/sql/SQLStorage.ts [hunk 1]
```diff
@@ -95,7 +95,9 @@ export class SQLStorage {
    * Checks if storage tables are created
    */
   public async isInitialized() {
-    return !this._checkMissingTables();
+    const hasMissingTables = await this._checkMissingTables();
+    this._isReady = !hasMissingTables;
+    return !hasMissingTables;
```
Comment (VictorVelarde) [inline packages/custom-storage/src/sql/SQLStorage.ts]:
```suggestion
    return this._isReady;
```

File: packages/custom-storage/src/CustomStorage.ts [hunk 2]
```diff
@@ -212,8 +212,10 @@ export class CustomStorage implements StorageRepository {
     await this._publicSQLStorage.destroy();
   }
 
-  private isInitialized() {
-    return this._publicSQLStorage.isInitialized();
+  public async isInitialized() {
+    const inits = await Promise.all([this._publicSQLStorage.isInitialized(), this._privateSQLStorage.isInitialized()]);
```
Comment (VictorVelarde) [inline packages/custom-storage/src/CustomStorage.ts]: Why do we need to check both storages?

File: packages/custom-storage/src/sql/SQLStorage.ts [hunk 3]
```diff
@@ -95,7 +95,9 @@ export class SQLStorage {
    * Checks if storage tables are created
    */
   public async isInitialized() {
-    return !this._checkMissingTables();
+    const hasMissingTables = await this._checkMissingTables();
+    this._isReady = !hasMissingTables;
```
Comment (VictorVelarde) [inline packages/custom-storage/src/sql/SQLStorage.ts]: ok, I see, this._isReady was not yet correctly setup...

File: packages/custom-storage/src/sql/SQLStorage.ts [hunk 4]
```diff
@@ -532,16 +534,23 @@ export class SQLStorage {
...
+        const checksTablesAreReady = requiredTables.map((table) => this._sql.query(`SELECT to_regclass('${table}')`));
+        const results = await Promise.all(checksTablesAreReady);
+        const missingTables = results.some((response: any) => {
+          const tableIsMissing = (response.rows[0].to_regclass === null);
+          return tableIsMissing;
+        });
+
+        resolve(missingTables);
+      } catch (err) {
+        reject(err);
```
Comment (VictorVelarde) [inline packages/custom-storage/src/sql/SQLStorage.ts]: Out of curiosity, what kind of errors have you found here?

Comment (neokore) [reply packages/custom-storage/src/sql/SQLStorage.ts]: No errors yet, but as long as I have to create a Promise, I always prefer to make use of the reject function for possible errors

File: packages/custom-storage/src/CustomStorage.ts (hunk 2 above)
Comment (neokore) [reply packages/custom-storage/src/CustomStorage.ts]: In order to set both as ready

**Pull request merged**